donkeycar/parts/tub_v2.py

```diff
@@ -105,21 +110,44 @@ class TubWriter(object):
     """
     def __init__(self, base_path, inputs=[], types=[], metadata=[],
                  max_catalog_len=1000):
-        self.tub = Tub(base_path, inputs, types, metadata, max_catalog_len)
+        self.tub = Tub(base_path, inputs + ['session_id'], types + ['str'],
+                       metadata, max_catalog_len)
+        self.session_id = self.create_session_id()
+
         def shutdown_hook():
             self.close()
 
         # Register hook
         atexit.register(shutdown_hook)
 
     def run(self, *args):
-        assert len(self.tub.inputs) == len(args)
-        record = dict(zip(self.tub.inputs, args))
+        assert len(self.tub.inputs) - 1 == len(args), \
+            f'Expected {len(self.tub.inputs)-1} inputs but received {len(args)}'
+        record = dict(zip(self.tub.inputs, args + (self.session_id,)))
         self.tub.write_record(record)
         return self.tub.manifest.current_index
 
     def __iter__(self):
         return self.tub.__iter__()
 
     def close(self):
-        self.tub.manifest.close()
+        self.tub.close()
+
+    def create_session_id(self):
```

Comment:
I don't like this idea at all.

We should be able to derive the name of the last session from the manifest metadata and we should not have to read a tub record for that. So add that to the manifest metadata.

Reply:
Looking into this.

Reply:
Thinking about this a bit more, we should move all the `Tub` specific information to a file called `tub.manifest` which is a `Seekable` as well or just a standard `file` with some JSON content. That might be a good way to decouple it from the actual `catalog` itself. Because the actual `session` information is orthogonal.

Reply:
Yeah we could as well add a line to the `manifest.json` - because the only thing we need is a list of `session_id`s.